microservices/netinfra_utils/workers/netconf_worker_test.py

```diff
@@ -0,0 +1,142 @@
+
+netconf_node_connecting = {
+    "node": [
+        {
+            "node-id": "xr6",
+            "netconf-node-topology:host": "192.168.1.213",
+            "netconf-node-topology:connected-message": "Connecting",
+            "netconf-node-topology:connection-status": "connecting",
+            "netconf-node-topology:port": 830
+        }
+    ]
+}
+
+netconf_node_connected = {
+    "node": [
+        {
+            "node-id": "xr6",
+            "netconf-node-topology:unavailable-capabilities": {
+                "unavailable-capability": []
+            },
+            "netconf-node-topology:available-capabilities": {
+                "available-capability": []
+            },
+            "netconf-node-topology:host": "192.168.1.213",
+            "netconf-node-topology:connected-message": "Success",
+            "netconf-node-topology:connection-status": "connected",
+            "netconf-node-topology:port": 830,
+            "topology-node-extension:node-type-fingerprint": "1319530689"
+        }
+    ]
+}
+
+netconf_node_non_exist = {
+    "errors": {
+        "error": [
+            {
+                "error-message": "Request could not be completed because the relevant data model content does not exist",
+                "error-tag": "data-missing",
+                "error-type": "protocol"
+            }
+        ]
+    }
+}
+
+
+class MockResponse:
+    def __init__(self, content, status_code):
+        self.content = content
+        self.status_code = status_code
+
+    def json(self):
+        return self.content
+
+
+class TestMount(unittest.TestCase):
+    def test_mount_new_device(self):
+        with patch('netconf_worker.requests.put') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps({}), encoding='utf-8'), 201)
+            request = netconf_worker.execute_mount_netconf(
+                {"inputData": {"device_id": "xr6", "host": "192.168.1.1", "port": "830", "keepalive-delay": "1000",
+                               "tcp-only": "false", "username": "name", "password": "password"}})
+            self.assertEqual(request["status"], "COMPLETED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf/node=xr6")
+            self.assertEqual(request["output"]["response_code"], 201)
+            self.assertEqual(request["output"]["response_body"], {})
+
+    def test_mount_existing_device(self):
+        with patch('netconf_worker.requests.put') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps({}), encoding='utf-8'), 204)
+            request = netconf_worker.execute_mount_netconf(
+                {"inputData": {"device_id": "xr6", "host": "192.168.1.1", "port": "830", "keepalive-delay": "1000",
+                               "tcp-only": "false", "username": "name", "password": "password"}})
+            self.assertEqual(request["status"], "FAILED")
```

Comment:
why the status is FAILED? 204 means that data was replaced in uniconfig.

Reply:
fixed